db/trie/branchnode.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2018 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package trie
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/logger"
+	"github.com/iotexproject/iotex-core/pkg/hash"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/proto"
+)
+
+const radix = 256
+
+type branchNode struct {
+	children map[byte]hash.Hash32B
+	ser      []byte
+}
+
+func newEmptyBranchNode() *branchNode {
+	return &branchNode{children: map[byte]hash.Hash32B{}}
+}
+
+func newBranchNodeFromProtoPb(pb *iproto.BranchPb) *branchNode {
+	b := newEmptyBranchNode()
+	for _, n := range pb.Branches {
+		b.children[byte(n.Index)] = byteutil.BytesTo32B(n.Path)
+	}
+	return b
+}
+
+func (b *branchNode) Children(ctx context.Context) ([]Node, error) {
+	tc, ok := getSameKeyLenTrieContext(ctx)
+	if !ok {
+		return nil, errors.New("failed to get operation context")
+	}
+	children := []Node{}
+	for i := range b.children {
+		if c, err := b.child(tc, i); err != nil {
+			return nil, err
+		} else if c != nil {
+			children = append(children, c)
+		}
+	}
+
+	return children, nil
+}
+
+func (b *branchNode) Type() NodeType {
+	return BRANCH
+}
+
+func (b *branchNode) Value() []byte {
+	return nil
+}
+
+func (b *branchNode) delete(tc SameKeyLenTrieContext, key keyType, offset uint8) (Node, error) {
+	logger.Debug().Hex("key", key[:]).Uint8("offset", offset).Msg("delete from a branch")
+	ok := key[offset]
```

Comment:
can you use a diff name? ok is a little confusing with the usage, v, ok := map[k]